colour/colorimetry/spectrum.py

```diff
@@ -3690,9 +3939,10 @@ def interpolate(self, shape=SpectralShape(), method=None):
 
     def align(self,
               shape,
-              method='Constant',
-              left=None,
-              right=None):
+              interpolation_method=None,
+              extrapolation_method='Constant',
+              extrapolation_left=None,
```

Comment:
Here are I think most of the callable that use a `*method` argument:

```python
    - Occurrences of 'method=' in Project with mask '*.py' (except comments and string literals)
- /Users/kelsolaar/Documents/Development/colour-science/colour/colour/adaptation
    - cmccat2000.py
        method='Forward'):
- /Users/kelsolaar/Documents/Development/colour-science/colour/colour/algebra
    - extrapolation.py
        method='Linear',
- /Users/kelsolaar/Documents/Development/colour-science/colour/colour/colorimetry
    - correction.py
    def bandpass_correction(spd, method='Stearns 1988'):
    - lefs.py
        method='MOVE',
        method='MOVE',
    - lightness.py
        def lightness(Y, method='CIE 1976', **kwargs):
    - luminance.py
        def luminance(LV, method='CIE 1976', **kwargs):
    - spectrum.py
        method='Constant',
        method=method, left=left, right=right)
        def interpolate(self, shape=SpectralShape(), method=None):
                  interpolation_method=None,
                  extrapolation_method='Constant',
        method='Constant',
        def interpolate(self, shape=SpectralShape(), method=None):
                  interpolation_method=None,
                  extrapolation_method='Constant',
    - tristimulus.py
        mi_5nm_omission_method=True,
        mi_20nm_interpolation_method=True):
        method='ASTM E308–15',
                      method=None):
    - whiteness.py
    d   ef whiteness(method='CIE 2004', **kwargs):
- /Users/kelsolaar/Documents/Development/colour-science/colour/colour/colorimetry/tests
    - tests_lefs.py
       method='LRC'),
       method='LRC'),
    - tests_spectrum.py
        method='Linear')[410],
        method='Pchip')[410],
        method='Linear')[411],
        method='Pchip')[411],
    - tests_tristimulus.py
        cmfs.shape, interpolation_method='Linear')
            mi_5nm_omission_method=False),
            mi_5nm_omission_method=False),
            mi_5nm_omission_method=False),
            mi_5nm_omission_method=False),
            mi_20nm_interpolation_method=False),
            mi_20nm_interpolation_method=False),
            mi_20nm_interpolation_method=False),
            mi_20nm_interpolation_method=False),
- /Users/kelsolaar/Documents/Development/colour-science/colour/colour/difference
    - delta_e.py
        def delta_E(Lab_1, Lab_2, method='CMC', **kwargs):
- /Users/kelsolaar/Documents/Development/colour-science/colour/colour/examples/adaptation
    - examples_cmccat2000.py
        XYZ_c, XYZ_w, XYZ_wr, L_A1, L_A2, method='Reverse'))
- /Users/kelsolaar/Documents/Development/colour-science/colour/colour/examples/algebra
    - examples_interpolation.py
       method='Pchip')
- /Users/kelsolaar/Documents/Development/colour-science/colour/colour/examples/colorimetry
    - examples_lightness.py
        print(colour.lightness(Y, method='Glasser 1958'))
        print(colour.lightness(Y, method='Wyszecki 1963'))
        print(colour.lightness(Y, method='CIE 1976'))
    - examples_luminance.py
        print(colour.luminance(V, method='Newhall 1943'))
        print(colour.luminance(V, method='ASTM D1535-08'))
    - examples_whiteness.py
        print(colour.whiteness(XYZ=XYZ, XYZ_0=XYZ_0, method='Berger 1959'))
        print(colour.whiteness(XYZ=XYZ, XYZ_0=XYZ_0, method='Taube 1960'))
        print(colour.whiteness(Lab=Lab, method='Stensby 1968'))
        print(colour.whiteness(XYZ=XYZ, method='ASTM 313'))
        print(colour.whiteness(xy=xy, Y=Y, method='Ganz 1979'))
- /Users/kelsolaar/Documents/Development/colour-science/colour/colour/examples/difference
    - examples_delta_e.py
        print(colour.delta_E(Lab1, Lab2, method='CIE 1976'))
        print(colour.delta_E(Lab1, Lab2, method='CIE 1994'))
        print(colour.delta_E(Lab1, Lab2, method='CIE 1994', textiles=False))
        print(colour.delta_E(Lab1, Lab2, method='CIE 2000'))
        print(colour.delta_E(Lab1, Lab2, method='CMC'))
        print(colour.delta_E(Lab1, Lab2, method='CMC', l=1))
- /Users/kelsolaar/Documents/Development/colour-science/colour/colour/examples/models
    - examples_conversion_functions.py
        print(colour.linear_to_log(linear, method='Cineon'))
        print(colour.log_to_linear(log, method='Cineon'))
        print(colour.linear_to_log(linear, method='Panalog'))
        print(colour.log_to_linear(log, method='Panalog'))
        print(colour.linear_to_log(linear, method='REDLogFilm'))
        print(colour.log_to_linear(log, method='REDLogFilm'))
        print(colour.linear_to_log(linear, method='ViperLog'))
        print(colour.log_to_linear(log, method='ViperLog'))
        print(colour.linear_to_log(linear, method='PLog'))
        print(colour.log_to_linear(log, method='PLog'))
        print(colour.linear_to_log(linear, method='C-Log'))
        print(colour.log_to_linear(log, method='C-Log'))
        print(colour.linear_to_log(linear, method='ACEScc'))
        print(colour.log_to_linear(log, method='ACEScc'))
        print(colour.linear_to_log(linear, method='Alexa Log C'))
        print(colour.log_to_linear(log, method='Alexa Log C'))
        print(colour.linear_to_log(linear, method='S-Log'))
        print(colour.log_to_linear(log, method='S-Log'))
        print(colour.linear_to_log(linear, method='S-Log2'))
        print(colour.log_to_linear(log, method='S-Log2'))
        print(colour.linear_to_log(linear, method='S-Log3'))
        print(colour.log_to_linear(log, method='S-Log3'))
- /Users/kelsolaar/Documents/Development/colour-science/colour/colour/examples/notation
    - examples_munsell.py
        print(colour.munsell_value(Y, method='Priest 1920'))
        print(colour.munsell_value(Y, method='Munsell 1933'))
        print(colour.munsell_value(Y, method='Moon 1943'))
        print(colour.munsell_value(Y, method='Saunderson 1944'))
        print(colour.munsell_value(Y, method='Ladd 1955'))
        print(colour.munsell_value(Y, method='McCamy 1987'))
        print(colour.munsell_value(Y, method='ASTM D1535-08'))
- /Users/kelsolaar/Documents/Development/colour-science/colour/colour/examples/plotting
    - examples_colorimetry_plots.py
        bandpass_corrected_street_light_spd, method='Stearns 1988')
        ASTM_G_173_spd.interpolate(colour.SpectralShape(interval=5), method='Linear')
- /Users/kelsolaar/Documents/Development/colour-science/colour/colour/examples/temperature
    - examples_cct.py
        print(colour.uv_to_CCT(uv, method='Robertson 1968'))
        print(colour.CCT_to_uv(CCT, D_uv, method='Robertson 1968'))
        print(colour.xy_to_CCT(xy, method='McCamy 1992'))
        print(colour.xy_to_CCT(xy, method='Hernandez 1999'))
        print(colour.CCT_to_xy(CCT, method="Kang 2002"))
        print(colour.CCT_to_xy(CCT, method="CIE Illuminant D Series"))
- /Users/kelsolaar/Documents/Development/colour-science/colour/colour/models/rgb
    - common.py
        chromatic_adaptation_method='CAT02',
    - conversion_functions.py
        def linear_to_log(value, method='Cineon', **kwargs):
        def log_to_linear(value, method='Cineon', **kwargs):
- /Users/kelsolaar/Documents/Development/colour-science/colour/colour/models/rgb/dataset
    - alexa_wide_gamut_rgb.py
       method='Linear Scene Exposure Factor',
       method='Linear Scene Exposure Factor',
- /Users/kelsolaar/Documents/Development/colour-science/colour/colour/models/rgb/tests
    - tests_common.py
        chromatic_adaptation_method='Bradford'),
- /Users/kelsolaar/Documents/Development/colour-science/colour/colour/notation
    - munsell.py
        def munsell_value(Y, method='ASTM D1535-08'):
- /Users/kelsolaar/Documents/Development/colour-science/colour/colour/plotting
    - temperature.py
        xy = np.array([UCS_uv_to_xy(CCT_to_uv(x, 0, method='Robertson 1968'))
            x0, y0 = UCS_uv_to_xy(CCT_to_uv(i, -0.025, method='Robertson 1968'))
            x1, y1 = UCS_uv_to_xy(CCT_to_uv(i, 0.025, method='Robertson 1968'))
        uv = np.array([CCT_to_uv(x, 0, method='Robertson 1968')
            u0, v0 = CCT_to_uv(i, -0.05, method='Robertson 1968')
            u1, v1 = CCT_to_uv(i, 0.05, method='Robertson 1968')
- /Users/kelsolaar/Documents/Development/colour-science/colour/colour/temperature
    - cct.py
        def uv_to_CCT(uv, method='Ohno 2013', **kwargs):
        def CCT_to_uv(CCT, D_uv=0, method='Ohno 2013', **kwargs):
        def xy_to_CCT(xy, method='McCamy 1992', **kwargs):
        def CCT_to_xy(CCT, method='Kang 2002'):
- /Users/kelsolaar/Documents/Development/colour-science/colour/colour/volume
    - rgb.py
        chromatic_adaptation_method='CAT02',
        chromatic_adaptation_method='CAT02',
```

Currently most of them (if not all) are kinda high level API arguments, taking either a string or boolean argument.